### PR TITLE
Add pagination to around page

### PR DESCRIPTION
**Pull request by dracos**
This factors this code to use the same code `/reports` and `/my` are using, and removes the show old pins code. For #525.
Also adds heading for #1577 (as those nearby ones are excluded from the pagination, which is a bit odd I grant you, but didn't seem odd enough to try and work something out).

### CODE REVIEW

File: perllib/FixMyStreet/DB/ResultSet/Problem.pm [hunk 1]
```diff
@@ -140,11 +140,11 @@ sub _recent {
 # Problems around a location
 
 sub around_map {
-    my ( $rs, $limit, %p) = @_;
+    my ( $rs, %p) = @_;
     my $attr = {
         order_by => $p{order},
     };
-    $attr->{rows} = $limit if $limit;
+    $attr->{rows} = FixMyStreet->config('ALL_REPORTS_PER_PAGE');
```
Comment (struan) [inline perllib/FixMyStreet/DB/ResultSet/Problem.pm]: Should this not use `cobrand->reports_per_page`?

File: templates/web/base/around/on_map_list_items.html [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+[% IF around_map.size %]
+<ul class="item-list item-list--reports">
+    [% FOREACH problem IN around_map %]
+        [% dist = tprintf("%.1f", (problem.distance || 0) );
+           problem = problem.problem;
+        %]
+        [% INCLUDE 'reports/_list-entry.html' %]
+    [% END %]
+</ul>
+[% END %]
```
Comment (struan) [inline templates/web/base/around/on_map_list_items.html]: This change seems to have broken the highlight pin on rollover behaviour.

File: perllib/FixMyStreet/App/Controller/Around.pm [hunk 3]
```diff
@@ -167,10 +167,7 @@ sub display_location : Private {
     my $latitude  = $c->stash->{latitude};
     my $longitude = $c->stash->{longitude};
 
-    # Deal with pin hiding/age
-    my $all_pins = $c->get_param('all_pins') ? 1 : undef;
-    $c->stash->{all_pins} = $all_pins;
-    my $interval = $all_pins ? undef : $c->cobrand->on_map_default_max_pin_age;
+    my $interval = $c->cobrand->on_map_default_max_pin_age;
```
Comment (struan) [inline perllib/FixMyStreet/App/Controller/Around.pm]: So how do I see really old reports now?

Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Around.pm]: Whoops, no need for that interval at all any more, is there.

File: templates/web/base/around/on_map_list_items.html (hunk 2 above)
Comment (dracos) [inline templates/web/base/around/on_map_list_items.html]: Turns out this is also already broken on e.g. a `/reports` page after you pan the map and the pins have updated. Will fix for everywhere hopefully.

File: perllib/FixMyStreet/DB/ResultSet/Problem.pm (hunk 1 above)
Comment (dracos) [inline perllib/FixMyStreet/DB/ResultSet/Problem.pm]: Annoying have to get the cobrand down in that fn now though, hmm.

**Comment by dracos** [comment]
Tested on live today for a while:
- Pagination expects ajax calls to be same as normal calls with ajax=1 query parameter. `/around` currently uses `/ajax`. This also means /around has to work with a bbox parameter, like the ajax version does.
- Pagination reload with bbox in URL does not remember zoom (it does get centre from the bbox parameter). This currently works on `/reports` because that has zoom to extent on load and no reload of pins on load. I think having zoom to extent on other pages would be okay, too, maybe.
- ajax currently ignores pagination (/reports etc pagination reloads full page); this means when the pins load on page load on /around they reset to page 1.

The pagination links should perform an ajax call rather than a full page load, and that would hopefully fix most/all the above.

**Comment by davea** [comment]
There's an odd bug which is stopping all pins being shown on the map when you zoom in.

If you go to a location and zoom in a couple of times, no more pins are added to the map:
zoom1

However if you change the filters once you're zoomed in to force a reload of all pins in the viewport, the 'hidden' pins are revealed:
zoom2

Here's a comparison of the two end results highlighting the difference:
comparison


I'm guessing this is due to some odd interaction between the pagination and the 'don't reload everything when zooming in' behaviours?

**Comment by dracos** [comment]
Ah yes, I need to behave as on `/reports` and refresh on zoom (as default strategy assumes you're showing everything within viewport so need do nothing on zoom in). Have tweaked default Strategy (that `/reports` strategy inherits from) to refresh on zoom, hopefully that fixes it.

**Pull request merged**